upt_macports/upt_macports.py

```diff
@@ -176,6 +180,47 @@ def create_package(self, upt_pkg, output=None):
             pkg_cls = pkg_classes[upt_pkg.frontend]
         except KeyError:
             raise upt.UnhandledFrontendError(self.name, upt_pkg.frontend)
-
         packager = pkg_cls()
         packager.create_package(upt_pkg, output)
+
+    def pkg_exist(self):
+        prefix = {
+            'pypi': 'py-',
+            'rubygems': 'rb-',
+            'cpan': 'p5-',
+        }
+        port_name = prefix.get(self.upt_pkg.frontend, '') + self.upt_pkg.name
+        self.logger.info(f'Checking MacPorts tree for port {port_name}')
+        cmd = f'port info --version {port_name}'
+        port = subprocess.getoutput(cmd)
```

Comment:
Does port(1) never fail? Isn't there some error-handling we'd like to do here?

Reply:
Even if `port` fails then that output will be stored in port which the case when port is not found:
`Error: Port portname not found`

Reply:
I assume that @Steap refers to the actual ```port``` command failing, for example because it is not installed, not in the path or something similar.

Reply:
ohh ok, then for that we can do something like:
```
if port.startswith('Error') 
 ...
elif port.startswith('version'):
 ...
else:
    `throw_an_error`
```

Reply:
Exactly, we should issue a Warning/Error that we cannot run the ```port``` command